### ISSUE TITLE
Ambassador 0.50.0-ea5 upstream certs are offered to client

### ISSUE THREAD

**Issue by n1koo** [datawire/ambassador#983]
**Describe the bug**
When using MTLS the certs that should be used for upstream are offered to the client.

Example config with Istio:

```
      apiVersion: ambassador/v0
      kind: Module
      name: tls
      config:
        server:
          enabled: true
          redirect_cleartext_from: 8080
          secret: ambassador-certs
        client:
          enabled: false
        upstream:
          cert_chain_file: /etc/istiocerts/cert-chain.pem
          private_key_file: /etc/istiocerts/key.pem
```

This is also somehow related to https://github.com/datawire/ambassador/issues/981 , eg., if I remove `redirect_cleartext_from: 8080` it works again. Doesn't matter if the http or https endpoint is hit. 

**To Reproduce**
Add `upstream` certs, hit any TLS enabled endpoint

**Expected behavior**
`server.secret` should be used to match correct cert

**Versions (please complete the following information):**
 - Ambassador: 0.50.0-ea5
 - Bare metal Kubernetes 1.12.0

**Additional context**

**Comment by containscafeine** [comment]
@n1koo this should have been fixed in #1037 #1042, can you confirm?

**Comment by kflynn** [comment]
Closing fixed in -rc1; @n1koo please reopen if you have trouble!
